internal/api/commitment.go

```diff
@@ -850,6 +858,180 @@ func (p *v1Provider) GetCommitmentConversions(w http.ResponseWriter, r *http.Req
+		http.Error(w, "no transfer token provided", http.StatusBadRequest)
+		return
+	}
+	dbDomain := p.FindDomainFromRequest(w, r)
+	if dbDomain == nil {
+		return
+	}
+	targetProject := p.FindProjectFromRequest(w, r, dbDomain)
+	if targetProject == nil {
+		return
+	}
+
+	// sourceBehavior
+	var dbCommitment db.ProjectCommitment
+	err := p.DB.SelectOne(&dbCommitment, `SELECT pc.* FROM project_commitments pc WHERE pc.id = $1`, commitmentID)
+	if errors.Is(err, sql.ErrNoRows) {
+		http.Error(w, "no such commitment", http.StatusNotFound)
+		return
+	} else if respondwith.ErrorText(w, err) {
+		return
+	}
+	var sourceLoc datamodel.AZResourceLocation
+	err = p.DB.QueryRow(findProjectAZResourceLocationByIDQuery, dbCommitment.AZResourceID).
+		Scan(&sourceLoc.ServiceType, &sourceLoc.ResourceName, &sourceLoc.AvailabilityZone)
+	if errors.Is(err, sql.ErrNoRows) {
+		// defense in depth: this should not happen because all the relevant tables are connected by FK constraints
+		http.Error(w, "no route to this commitment", http.StatusNotFound)
+		return
+	} else if respondwith.ErrorText(w, err) {
+		return
+	}
+	sourceBehavior := p.Cluster.BehaviorForResource(sourceLoc.ServiceType, sourceLoc.ResourceName)
+	if len(sourceBehavior.CommitmentDurations) == 0 {
+		http.Error(w, "commitments are not enabled for this resource", http.StatusUnprocessableEntity)
+		return
+	}
+	if dbCommitment.ConfirmBy != nil {
+		http.Error(w, "unable to convert unconfirmed commitments", http.StatusConflict)
+		return
+	}
+
+	// targetBehavior
+	var parseTarget struct {
+		Request struct {
+			TargetService  limes.ServiceType           `json:"target_service"`
+			TargetResource limesresources.ResourceName `json:"target_resource"`
+			SourceAmount   uint64                      `json:"source_amount"`
+			TargetAmount   uint64                      `json:"target_amount"`
+		} `json:"commitment"`
+	}
+	if !RequireJSON(w, r, &parseTarget) {
+		return
+	}
+	req := parseTarget.Request
+	nm := core.BuildNameMapping(p.Cluster)
+	targetServiceType, targetResourceName, exists := nm.MapResourceFromV1API(req.TargetService, req.TargetResource)
+	if !exists {
+		msg := fmt.Sprintf("no such service and/or resource: %s/%s", req.TargetService, req.TargetResource)
+		http.Error(w, msg, http.StatusUnprocessableEntity)
+		return
+	}
+	targetBehavior := p.Cluster.BehaviorForResource(targetServiceType, targetResourceName)
+	if len(targetBehavior.CommitmentDurations) == 0 {
+		msg := fmt.Sprintf("commitments are not enabled for resource %s/%s", req.TargetService, req.TargetResource)
+		http.Error(w, msg, http.StatusUnprocessableEntity)
+		return
+	}
+	if sourceBehavior.CommitmentConversion.Identifier == "" || sourceBehavior.CommitmentConversion.Identifier != targetBehavior.CommitmentConversion.Identifier {
+		http.Error(w, "commitment is not convertible", http.StatusUnprocessableEntity)
+		return
+	}
+
+	// conversion
+	if req.SourceAmount <= 0 || req.SourceAmount > dbCommitment.Amount {
```

Comment:
SourceAmount is unsigned:
```suggestion
	if req.SourceAmount > dbCommitment.Amount {
```